service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -82,107 +126,147 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
+        runStateResponseList.stream()
+            .filter(run -> CbasRunStatus.fromValue(run.getState()).equals(SYSTEM_ERROR))
+            .toList();
 
-    UUID runId = UUID.randomUUID();
-    String dataTableRowId = request.getWdsRecords().getRecordIds().get(0);
+    if (runsInErrorState.size() == recordIds.size()) {
+      runSetState = ERROR;
+    } else runSetState = RUNNING;
 
-    // Submit the workflow and get its ID:
-    RunId workflowResponse;
+    // Return the result
+    return new ResponseEntity<>(
+        new RunSetStateResponse()
+            .runSetId(runSetId.toString())
+            .runs(runStateResponseList)
+            .state(runSetState),
+        HttpStatus.OK);
+  }
 
-    try {
-      workflowResponse = cromwellService.submitWorkflow(request.getWorkflowUrl(), params);
-    } catch (cromwell.client.ApiException e) {
-      log.warn("Cromwell submission failed. ApiException", e);
-      String errorMessage;
-      errorMessage = e.getMessage();
-      runDao.createRun(
-          new Run(
-              runId,
-              null,
-              runSet,
-              dataTableRowId,
-              OffsetDateTime.now(),
-              SYSTEM_ERROR,
-              OffsetDateTime.now(),
-              OffsetDateTime.now(),
-              errorMessage));
-      // TODO: After WM-1448 completes, add nicer-looking return response that returns
-      //  the error to the user immediately
-      return new ResponseEntity<>(HttpStatus.valueOf(e.getCode()));
-    } catch (JsonProcessingException e) {
-      // Should be super rare that jackson cannot convert an object to Json...
-      log.warn("Failed to convert inputs object to JSON", e);
-      return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+  public static Optional<ResponseEntity<RunSetStateResponse>> validateRequest(
+      RunSetRequest request, int maxRecordIds) {
+    String errorMsg = "";
+
+    // check number of Record IDs in request is within allowed limit
+    int recordIdsSize = request.getWdsRecords().getRecordIds().size();
+    if (recordIdsSize > maxRecordIds) {
+      errorMsg =
+          "%s Record IDs submitted exceeds the maximum value of %s. "
+              .formatted(recordIdsSize, maxRecordIds);
+    }
+
+    // check that there are no duplicated Record IDs present in the request
+    List<String> recordIds = request.getWdsRecords().getRecordIds();
+    List<String> duplicateRecordIds =
+        recordIds.stream().filter(e -> Collections.frequency(recordIds, e) > 1).distinct().toList();
+    if (duplicateRecordIds.size() > 0) {
+      errorMsg +=
+          String.format("Duplicate Record ID(s) %s present in request.", duplicateRecordIds);
+    }
+
+    if (!errorMsg.isEmpty()) {
+      String finalErrorMsg = "Bad user request. Error(s): " + errorMsg;
+      log.warn(finalErrorMsg);
+      return Optional.of(
+          new ResponseEntity<>(
+              new RunSetStateResponse().errors(finalErrorMsg), HttpStatus.BAD_REQUEST));
```

Comment:
After talking with Chris and Michael, we decided to return a `List<String>` to return the errors instead of `Optional<ResponseEntity<RunSetStateResponse>>` so that it is an addition of previous method rather than removal.